src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -909,6 +906,10 @@ def _exec(
                 run_info.inputs = inputs
             output, nodes_outputs = self._traverse_nodes(inputs, context)
             output = self._stringify_generator_output(output) if not allow_generator_output else output
+            # Persist the node runs for the nodes that have a generator output
+            run_tracker.persist_selected_node_runs(
+                run_info, [nodename for nodename, output in nodes_outputs.items() if isinstance(output, GeneratorType)]
```

Comment:
We can extract a variable here for better understanding.
i.e. `nodes_that_return_generator` (Think for a better name)

Reply:
We can use `(` to replace `[` here too.